src/savegame.cpp

```diff
@@ -635,69 +643,1183 @@ void ingame_savegame::write_game(config_writer &out) {
+	ss << "                    [/not]\n";
+	ss << "                    [not]\n";
+	ss << "                        [status]\n";
+	ss << "                            fighting=yes\n";
+	ss << "                        [/status]\n";
+	ss << "                    [/not]\n";
+	ss << "                    [modifications]\n";
+	ss << "                        [trait]\n";
+	ss << "                            id=healthy_1_4\n";
+	ss << "                        [/trait]\n";
+	ss << "                    [/modifications]\n";
+	ss << "                [/filter_wml]\n";
+	ss << "            [/filter]\n";
+	ss << "            amount=2\n";
+	ss << "            animate=yes\n";
+	ss << "            restore_statuses=no\n";
+	ss << "        [/heal_unit]\n";
+	read(event, ss);
+}
+
+void healthy_trait_1_4_turn_refresh(config& event)
+{
+	std::stringstream ss;
+	ss << "        name=\"healthy turn refresh\"\n";
+	ss << "        name=\"turn refresh\"\n";
+	ss << "        first_time_only=no\n";
+	ss << "        [modify_unit]\n";
+	ss << "            [filter]\n";
+	ss << "                side=$side_number\n";
+	ss << "            [/filter]\n";
+	ss << "            [object]\n";
+	ss << "            [effect]\n";
+	ss << "                apply_to=status\n";
+	ss << "                remove=fighting\n";
+	ss << "            [/effect]\n";
+	ss << "            [/object]\n";
+	ss << "        [/modify_unit]\n";
+	read(event, ss);
+}
+
+void healthy_trait_1_4_attack_end(config& event)
+{
+	std::stringstream ss;
+	ss << "        name=\"healthy attack end\"\n";
+	ss << "        name=\"attack end\"\n";
+	ss << "        first_time_only=no\n";
+	ss << "        [modify_unit]\n";
+	ss << "            [filter]\n";
+	ss << "                id=$unit.id\n";
+	ss << "            [/filter]\n";
+	ss << "            [object]\n";
+	ss << "                [effect]\n";
+	ss << "                    apply_to=status\n";
+	ss << "                    add=fighting\n";
+	ss << "                [/effect]\n";
+	ss << "            [/object]\n";
+	ss << "        [/modify_unit]\n";
+	ss << "        [modify_unit]\n";
+	ss << "            [filter]\n";
+	ss << "                id=$second_unit.id\n";
+	ss << "            [/filter]\n";
+	ss << "            [object]\n";
+	ss << "                [effect]\n";
+	ss << "                    apply_to=status\n";
+	ss << "                    add=fighting\n";
+	ss << "                [/effect]\n";
+	ss << "            [/object]\n";
+	ss << "        [/modify_unit]\n";
+	read(event, ss);
+}
+
+void update_prestart_detect_factions(config& cfg)
+{
+	if(config& replay = cfg.child("replay_start"))
+	{
+		for(config& event: replay.child_range("event"))
+		{
+			if(event["name"] == "prestart")
+			{
+				if(config& fire_event = event.child("fire_event"))
+				{
+					if(fire_event["name"] == "place_units")
+					{
+						config& lua = event.child("lua");
+
+						std::string changes[] = {"if searched == actual then;if actual:sub(1, #searched) == searched  then",
```

Comment:
This seems like a place where extracting to an external .lua file would be more readable, though I'm not quite sure what that file would look like…